tests/unit_tests/test_nwbfile_metadata.py

```diff
@@ -317,11 +330,70 @@ def test_check_subject_age_iso8601():
     )
 
 
-def test_pass_check_subject_age_with_dob():
-    subject = Subject(subject_id="001", sex="Male", date_of_birth=datetime.now())
+def test_check_subject_age_iso8601_range_pass_1():
+    subject = Subject(subject_id="001", sex="Male", age="P1D/P3D")
+    assert check_subject_age(subject) is None
+
+
+def test_check_subject_age_iso8601_range_pass_2():
+    subject = Subject(subject_id="001", sex="Male", age="P1D/")
     assert check_subject_age(subject) is None
 
 
+def test_check_subject_age_iso8601_range_fail_1():
+    subject = Subject(subject_id="001", sex="Male", age="9 months/12 months")
+    assert check_subject_age(subject) == InspectorMessage(
+        message=(
+            "Subject age, '9 months/12 months', does not follow ISO 8601 duration format, e.g. 'P2Y' for 2 years "
+            "or 'P23W' for 23 weeks. You may also specify a range using a '/' separator, e.g., 'P1D/P3D' for an "
+            "age range somewhere from 1 to 3 days. If you cannot specify the upper bound of the range, "
+            "you may leave the right side blank, e.g., 'P90Y/' means 90 years old or older."
+        ),
+        importance=Importance.BEST_PRACTICE_SUGGESTION,
+        check_function_name="check_subject_age",
+        object_type="Subject",
+        object_name="subject",
+        location="/general/subject",
+    )
+
+
+def test_check_subject_age_iso8601_range_fail_2():
+    subject = Subject(subject_id="001", sex="Male", age="9 months/")
+    assert check_subject_age(subject) == InspectorMessage(
+        message=(
+            "Subject age, '9 months/', does not follow ISO 8601 duration format, e.g. 'P2Y' for 2 years "
+            "or 'P23W' for 23 weeks. You may also specify a range using a '/' separator, e.g., 'P1D/P3D' for an "
+            "age range somewhere from 1 to 3 days. If you cannot specify the upper bound of the range, "
+            "you may leave the right side blank, e.g., 'P90Y/' means 90 years old or older."
+        ),
+        importance=Importance.BEST_PRACTICE_SUGGESTION,
+        check_function_name="check_subject_age",
+        object_type="Subject",
+        object_name="subject",
+        location="/general/subject",
+    )
+
+
+def test_check_subject_proper_age_range_pass():
+    subject = Subject(subject_id="001", sex="Male", age="P1D/P3D")
+    assert check_subject_proper_age_range(subject) is None
+
+
+def test_check_subject_proper_age_range_fail():
+    subject = Subject(subject_id="001", sex="Male", age="P3D/P1D")
+    assert check_subject_proper_age_range(subject) == InspectorMessage(
+        message=(
+            f"The durations of the Subject age range, '{subject.age}', are not strictly increasing. "
```

Comment:
```suggestion
            "The durations of the Subject age range, 'P3D/P1D', are not strictly increasing. "
```

I like to have literal strings in the tests to make sure that the stringification of values is done as expected